Sources/System/Process/ResourceUsageInfo.swift

```diff
@@ -0,0 +1,159 @@
+
+extension ProcessID {
+  public struct ResourceUsageInfo: RawRepresentable/*, Hashable, Codable*/ {
+    /// The raw C process id.
+    @_alwaysEmitIntoClient
+    public let rawValue: CInterop.RUsageInfo
+
+    /// Creates a strongly-typed process id from a raw C pid
+    @_alwaysEmitIntoClient
+    public init(rawValue: CInterop.RUsageInfo) { self.rawValue = rawValue }
+
+    fileprivate init(_ rawValue: CInterop.RUsageInfo) {
+      self.init(rawValue: rawValue)
+    }
+
+    fileprivate static var blank: ResourceUsageInfo {
+      ResourceUsageInfo(rusage_info_current())
+    }
+  }
+}
+
+// FIXME(DO NOT MERGE): system_foo wrappers for these and mocking
+import CSystem
+extension ProcessID {
+  public func getResourceUsageInfo() throws -> ResourceUsageInfo {
+    var current = ResourceUsageInfo.blank
+    try withUnsafeMutablePointer(to: &current) {
+      try $0.withMemoryRebound(to: rusage_info_t?.self, capacity: 1) {
+        guard 0 == proc_pid_rusage(self.rawValue, RUSAGE_INFO_CURRENT, $0) else {
+          throw Errno(rawValue: errno)
+        }
+      }
+    }
+    return current
+  }
+}
+
+// FIXME: docs or comments, the headers have none...
+// FIXME: names
+extension ProcessID.ResourceUsageInfo {
+  // FIXME: UUID proper type
+  public typealias UUID = (
```

Comment:
Foundation already has a wrapper for `uuid_t` -- perhaps we should investigate shrinking it into System.